pkgs/development/compilers/ghc/8.2.1-binary.nix

```diff
@@ -4,7 +4,7 @@
 }:
 
 # Prebuilt only does native
-assert stdenv.targetPlatform == stdenv.hostPlatform;
+# assert stdenv.targetPlatform == stdenv.hostPlatform;
```

Comment:
While disabling this check, allows most of the cross compilation to succeed on macOS, I believe this is also the reason why we end up with mixed (build and target) packages in the stdenv eventually.